pkg/operator/controller/gatewayapi/controller.go

```diff
@@ -25,7 +27,10 @@ import (
 )
 
 const (
-	controllerName = "gatewayapi_controller"
+	controllerName                        = "gatewayapi_controller"
+	experimentalGatewayAPIGroupName       = "gateway.networking.x-k8s.io"
```

Comment:
nit: it would be nice to define this in names.go so it could be used by other packages.

Reply:
Currently it's only used in this package. In the unit tests, I deliberately abstain from using the same constants as in the code to avoid duplicating errors.